pkgs/by-name/li/libchewing/package.nix

```diff
@@ -4,31 +4,52 @@
   fetchFromGitHub,
   cmake,
   sqlite,
+  corrosion,
+  rustPlatform,
+  cargo,
+  rustc,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "libchewing";
-  version = "0.6.0";
+  version = "0.9.1";
 
   src = fetchFromGitHub {
     owner = "chewing";
     repo = "libchewing";
-    rev = "v${finalAttrs.version}";
-    sha256 = "sha256-X+4Rr5Mfc4qeJxmHczu4MKgHBvQN1rhqUrJSx8SFnDk=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-5aeAsvTiUMTm+ibNfJI57rzSUpJB7luhA/aWmTcnBj4=";
   };
 
-  buildInputs = [ sqlite ];
+  # ld: unknown option: -version-script
+  postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    substituteInPlace CMakeLists.txt \
+      --replace-fail "if(CMAKE_C_COMPILER_ID MATCHES GNU|^Clang)" "if((CMAKE_C_COMPILER_ID MATCHES GNU|^Clang) AND NOT APPLE)"
```

Comment:
Such condition guards four operations in the upstream `CMakeList.txt`. It might be worth investigating which of them causes the build failure on Darwin and replace the very line.

Nevertheless, I'll merge it first considering the result builds on both Linux and Darwin.